metals/src/main/scala/scala/meta/internal/metals/Messages.scala

```diff
@@ -703,7 +703,18 @@ object Messages {
     ): String = {
       val using = "legacy " + usingString(usingNow)
       val recommended = recommendationString(usingNow)
-      s"You are using $using, which might not be supported in future versions of Metals. " +
+      s"You are using $using, which might stop being bugfixed in future versions of Metals. " +
+        s"Please upgrade to $recommended."
```

Comment:
Would it be possible to print this message only when the user effectively uses a version whose support is not maintained anymore in Metals?

Reply:
Oh, maybe that’s what you did in `DeprecatedRemovedScalaVersion`. Then I would suggest not showing a warning to the user if everything works fine.

Reply:
We could remove that warning for sure, I was actually thinking about it. What do others think?

Reply:
I guess I'm a bit torn about it. I see the usefulness of telling a user that they're on a deprecated version and if so, they should update.

@julienrf are you thinking courses here? Is that what brings this up? You don't want students to see this?

Reply:
A counterpoint to this is that if users don't know they are using old things, they'll continue to use old things. If they get prompted to update, there is a good chance they'll update. Most of the time that update _will_ give them a smoother experience.

Reply:
This is not specifically about courses. I am not sure the role of the IDE should be to tell you to update your Scala version (unless that Scala version is not supported by the IDE, of course). I think the role of the IDE is to let the user work with the version of Scala they have to work with, without complaining :) Tools like dependabot or Scala Steward are for keeping things up to date and handling security issues. There could be an action in the IDE “check for dependency updates or security issues”, but I believe that should be an explicit action.

(I don’t feel super strongly about all that, it’s just that that warning “in the future, there _might_ not be bugfixes for the Metals support of this Scala version” looks pretty hypothetical, I don’t think it is worth notifying the user about that).

Reply:
>  I am not sure the role of the IDE should be to tell you to update your Scala version (unless that Scala version is not supported by the IDE, of course)

Sure, I can see this point. However, this is what Metals is saying here right? This version is deprecated _in Metals_? If we don't show any sort of warning, then there is really no reason to actually mark these as "deprecated". I don't feel strongly about this at all, but if we do mark them as deprecated we may want to notify the user of that. Again, I'm open to not, I don't have strong feelings on it.

Reply:
Ah, I might have misunderstood then, I thought we were showing a “pre-warning” when the version was still supported by Metals, but could hypothetically become unsupported in the future.

What are the messages we show to the user and when?

Reply:
There are a ton of different messages, but the relevant ones we're dealing with here are when we mark something as deprecated. So for example `3.3.0-RC1` had an issue, so we really don't want people on it. So if you use it right now with the latest snapshot you'd see something like this (although it differs per editor and this is nvim):
<img width="665" alt="Screenshot 2023-02-02 at 14 42 46" src="https://user-images.githubusercontent.com/13974112/216341016-1a78c6b4-83c7-4a38-a9f3-de9ae1e5a2a9.png">

Reply:
For instance, let’s say someone uses 2.13.2. Which message do they see, and why?

Reply:
They'd see the same message as above. Basically any Scala 2 version listed in https://github.com/scalameta/metals/blob/a20ca6d7b0e324aac552fbbbfa700804eec76752/project/V.scala#L66-L76
 or Scala 3 version listed in 
https://github.com/scalameta/metals/blob/a20ca6d7b0e324aac552fbbbfa700804eec76752/project/V.scala#L98 you'll see the same warning since they are deprecated in Metals, meaning we don't run tests against them.

Reply:
OK, so the reason here to show a warning is because that version of Scala stopped receiving updates in Metals support?

It seems different from what this message says:

> You are using x.y.z, which might stop being bugfixed in future versions of Metals.

Here it seems that we warn that in the future that version may stop receiving updates in Metals support. Maybe it’s just the message that could be improved? But IMHO, as a user I should not see a warning if everything works in the current version of Metals.

Reply:
Should we not show any warnings at all? The idea was to prompt users to update since they might not even know they have an old Scala version.

Currently, deprecated versions are not tested, so we are not really expecting this will be the best experience. To get the best experience you should update. There is almost no reason not to update your path version of Scala.

Reply:
In practice I think bumping Scala versions (even patch versions) requires some effort and may not always be prioritized, so I don’t think we should insist with that unless we know the IDE experience will be bad.

Reply:
We typically have the latest 3_ish_ versions of 2.12, 2.13 non-deprecated. So I still think if you're 4 patch versions behind a nudge isn't a bad thing.

Reply:
Let's keep the messages and we can change them if users complain too much.

Reply:
I guess now thinking about what @kpodsiad mentioned, I do think it's important here to make it clear that this is about metals. What do you think about:

```suggestion
      s"You are using $using, which has deprecated support in Metals meaning you're missing out on the latest features and bugfixes. " +
        s"Please upgrade to $recommended."
```

Reply:
Ach, right! I forgot about these messages, I made them similar to the ones in ProblemResolver